sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -395,6 +395,123 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       predicate)
   }
 
+  private def withAssignments(
+      assignCtx: SqlBaseParser.AssignmentListContext): (Seq[Expression], Seq[Expression]) =
+    withOrigin(assignCtx) {
+      assignCtx.assignment().asScala.map { assign =>
+        (UnresolvedAttribute(visitQualifiedName(assign.key)), expression(assign.value))
+      }.foldLeft(Seq[Expression](), Seq[Expression]()) {
+        case (a, b) => (a._1 :+ b._1, a._2 :+ b._2)
+      }
+    }
+
+  override def visitMergeIntoTable(ctx: MergeIntoTableContext): LogicalPlan = withOrigin(ctx) {
+    val targetTable = UnresolvedRelation(visitMultipartIdentifier(ctx.target))
+    val targetTableAlias = if (ctx.targetAlias != null) {
+      val ident = ctx.targetAlias.strictIdentifier()
+      // We do not allow columns aliases after table alias.
```

Comment:
this appears many times, even for UPDATE/DELETE. Shall we have a method to do it?

Reply:
done